src/hotspot/cpu/aarch64/aarch64_vector.ad

```diff
@@ -3006,56 +3006,119 @@ instruct reduce_addD_masked(vRegD dst_src1, vReg src2, pRegGov pg) %{
 
 // ------------------------------ Vector reduction mul -------------------------
 
-instruct reduce_mulI(iRegINoSp dst, iRegIorL2I isrc, vReg vsrc,
-                     vReg tmp1, vReg tmp2) %{
+instruct reduce_mulI_le128b(iRegINoSp dst, iRegIorL2I isrc, vReg vsrc,
+                            vReg tmp1, vReg tmp2) %{
   predicate(Matcher::vector_length_in_bytes(n->in(2)) == 8 ||
             Matcher::vector_length_in_bytes(n->in(2)) == 16);
   match(Set dst (MulReductionVI isrc vsrc));
   effect(TEMP_DEF dst, TEMP tmp1, TEMP tmp2);
-  format %{ "reduce_mulI $dst, $isrc, $vsrc\t# vector (64/128 bits). KILL $tmp1, $tmp2" %}
+  format %{ "reduce_mulI_le128b $dst, $isrc, $vsrc\t# vector (64/128 bits). KILL $tmp1, $tmp2" %}
   ins_encode %{
     BasicType bt = Matcher::vector_element_basic_type(this, $vsrc);
     uint length_in_bytes = Matcher::vector_length_in_bytes(this, $vsrc);
-    __ neon_reduce_mul_integral($dst$$Register, bt, $isrc$$Register,
-                                $vsrc$$FloatRegister, length_in_bytes,
-                                $tmp1$$FloatRegister, $tmp2$$FloatRegister);
+    __ reduce_mul_integral_le128b($dst$$Register, bt, $isrc$$Register,
+                                  $vsrc$$FloatRegister, length_in_bytes,
+                                  $tmp1$$FloatRegister, $tmp2$$FloatRegister);
   %}
   ins_pipe(pipe_slow);
 %}
 
-instruct reduce_mulL(iRegLNoSp dst, iRegL isrc, vReg vsrc) %{
+instruct reduce_mulI_gt128b(iRegINoSp dst, iRegIorL2I isrc, vReg vsrc,
+                            vReg tmp1, vReg tmp2, pRegGov pgtmp) %{
+  predicate(Matcher::vector_length_in_bytes(n->in(2)) > 16);
+  match(Set dst (MulReductionVI isrc vsrc));
+  effect(TEMP_DEF dst, TEMP tmp1, TEMP tmp2, TEMP pgtmp);
+  format %{ "reduce_mulI_gt128b $dst, $isrc, $vsrc\t# vector (> 128 bits). KILL $tmp1, $tmp2, $pgtmp" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this, $vsrc);
+    uint length_in_bytes = Matcher::vector_length_in_bytes(this, $vsrc);
+    __ reduce_mul_integral_gt128b($dst$$Register, bt, $isrc$$Register,
+                                  $vsrc$$FloatRegister, length_in_bytes,
+                                  $tmp1$$FloatRegister, $tmp2$$FloatRegister,
+                                  $pgtmp$$PRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct reduce_mulL_le128b(iRegLNoSp dst, iRegL isrc, vReg vsrc) %{
```

Comment:
I suggest using `_128b` here since only `2L` is matched here.

```suggestion
instruct reduce_mulL_128b(iRegLNoSp dst, iRegL isrc, vReg vsrc) %{
```